docker/lib/dependabot/docker/update_checker.rb

```diff
@@ -105,12 +95,15 @@ def version_up_to_date?
       end
 
       def version_tag_up_to_date?(version)
-        return unless version&.match?(NAME_WITH_VERSION)
+        return unless version
+
+        version_tag = Tag.new(version)
+        return unless version_tag.comparable?
 
-        latest_version = fetch_latest_version(version)
+        latest_tag = fetch_latest_version(version_tag)
 
-        old_v = numeric_version_from(version)
-        latest_v = numeric_version_from(latest_version)
+        old_v = version_tag.numeric_version
+        latest_v = latest_tag.numeric_version
 
         version_class.new(latest_v) <= version_class.new(old_v)
```

Comment:
Maybe delete these two intermediate vars? I think it'd be fine to inline stuff into the next line:
```suggestion
        version_class.new(latest_tag.numeric_version) <= version_class.new(current_tag.numeric_version)
```

Note that my suggestion includes a renaming of `version_tag` to `current_tag` so won't work as-is...

Reply:
Yeah, specially those `old_v` names seem unnecessary. I can do this as a followup PR since I will also probably memoize this values when first called to not run the same regex over an over again after inlining the code.

Reply:
Makes sense. 👍 for shipping as-is, but let's leave these threads about future cleanups unresolved so they're easy to spot later when you come back for the follow-on.